### PR TITLE
feat: redesign events 2

### PR DESCRIPTION
**Pull request by ryanseee**
https://www.pivotaltracker.com/story/show/177811023

### CODE REVIEW

File: cms/event/digitalise-your-supply-chains-to-save-cost-improve-efficiency-overall-profitability-with-trames.md [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 title: Digitalise Your Supply Chains to save cost, improve efficiency & Overall
   Profitability with Trames
+thumbnail: https://images.unsplash.com/photo-1551818255-e6e10975bc17?ixid=MnwxMjA3fDB8MHxzZWFyY2h8Mnx8Y29uZmVyZW5jZXxlbnwwfHwwfHw%3D&ixlib=rb-1.2.1&w=1000&q=80
 description: Designed for businesses of all sizes, learn how you can digitise
   your supply chains to save cost, improve efficiency and overall profitability
   by using the Trames platform.
 link: https://trames.sg/registration
-date: 8 Apr 2021
+date: 1 Dec 2021
 timeStart: 2021-03-19T07:00:00.000Z
```
Comment (simboonlong) [inline cms/event/digitalise-your-supply-chains-to-save-cost-improve-efficiency-overall-profitability-with-trames.md]: why date change?

File: src/components/UI/ResourceEvent/ResourceEvent.tsx [hunk 2]
```diff
@@ -19,92 +20,102 @@ export interface EventProps {
...
+      <div
+        className={`w-full bg-white px-5 pt-3 pb-5 ${
+          thumbnail ? "rounded-b-xl md:w-8/12 md:rounded-none md:rounded-r-xl" : "rounded-xl"
+        }`}
+      >
+        {/* {isFuture(new Date(date)) && (
+          <div className="border-2 border-orange rounded inline-block py-1 px-2 uppercase font-bold text-xs text-orange my-2">
+            Upcoming
+          </div>
+        )} */}
```
Comment (simboonlong) [inline src/components/UI/ResourceEvent/ResourceEvent.tsx]: can remove

File: src/pages/event/event.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  }
+`;
+
+const getSortedByDateDesc = (items: any[]) => {
+  items.sort((a, b): number => {
+    return compareDesc(new Date(a.attributes.date), new Date(b.attributes.date));
+  });
+
+  return items;
+};
```
Comment (simboonlong) [inline src/pages/event/event.tsx]: shift this to a common utility area, then import and use from there?

File: src/pages/event/event.tsx [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+export const filterByCategory = (item: string, allEvents: EventProps[]): EventProps[] => {
+  switch (true) {
+    case item === "All":
+      return allEvents;
```
Comment (simboonlong) [inline src/pages/event/event.tsx]: maybe can omit `All` since default is that case

File: src/pages/event/event.tsx [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    default:
+      return allEvents;
+  }
+};
+
+events = getSortedByDateDesc(events);
+
+export const EventPage: FunctionComponent = () => {
+  const categories = ["All", "Upcoming Event", "Past Event"];
+  const [category, setCategory] = useState("All");
```
Comment (simboonlong) [inline src/pages/event/event.tsx]: lets enum the `categories`? so its:

```
enum SomeName {
  ALL = "ALL",
  UPCOMING_EVENT = "UPCOMING_EVENT",
  PAST_EVENT = "PAST_EVENT",
}
```

File: src/tailwind.css [hunk 6]
```diff
@@ -1,3 +1,5 @@
+@import url("https://fonts.googleapis.com/css2?family=Ubuntu:ital,wght@0,300;0,400;0,500;0,700;1,300;1,400;1,500;1,700&display=swap");
+
```
Comment (simboonlong) [inline src/tailwind.css]: can shift to public index.html instead?

File: public/admin/config.yml [hunk 7]
```diff
@@ -61,6 +61,7 @@ collections:
       preview: false
     fields:
       - { label: "Title", name: "title", widget: "string", hint: "Title of event." }
+      - { label: "Featured Image", name: "thumbnail", widget: "image", required: false }
       - { label: "Description", name: "description", widget: "text", hint: "Short description of event." }
```
Comment (simboonlong) [inline public/admin/config.yml]:
```suggestion
      - { label: "Title", name: "title", widget: "string", hint: "Title of event." }
      - { label: "Thumbnail", name: "thumbnail", widget: "image", required: false }
      - { label: "Description", name: "description", widget: "text", hint: "Short description of event." }
```

File: cms/event/digitalise-your-supply-chains-to-save-cost-improve-efficiency-overall-profitability-with-trames.md (hunk 1 above)
Comment (ryanseee) [reply cms/event/digitalise-your-supply-chains-to-save-cost-improve-efficiency-overall-profitability-with-trames.md]: oh i wanted to show the register button design. ok i changed it back and created another dummy event for testing future events.

File: src/components/UI/ResourceEvent/ResourceEvent.tsx (hunk 2 above)
Comment (ryanseee) [reply src/components/UI/ResourceEvent/ResourceEvent.tsx]: done.

File: src/pages/event/event.tsx (hunk 3 above)
Comment (ryanseee) [reply src/pages/event/event.tsx]: done. shifted it to src/utils/index.ts.

File: src/pages/event/event.tsx (hunk 4 above)
Comment (ryanseee) [reply src/pages/event/event.tsx]: done.

File: src/pages/event/event.tsx (hunk 5 above)
Comment (ryanseee) [reply src/pages/event/event.tsx]: ok done.

File: src/tailwind.css (hunk 6 above)
Comment (ryanseee) [reply src/tailwind.css]: ok shifted to public/index.html. but i dont intend to merge this into master though, cause this one should eventually be in tradetrust-ui-components so both creator and tt can use this font.

File: public/admin/config.yml (hunk 7 above)
Comment (ryanseee) [reply public/admin/config.yml]: done.

File: cms/event/digitalise-your-supply-chains-to-save-cost-improve-efficiency-overall-profitability-with-trames.md (hunk 1 above)
Comment (simboonlong) [reply cms/event/digitalise-your-supply-chains-to-save-cost-improve-efficiency-overall-profitability-with-trames.md]: maybe storybook?

File: src/tailwind.css (hunk 6 above)
Comment (simboonlong) [reply src/tailwind.css]: changing to used fonts weights only

File: cms/event/digitalise-your-supply-chains-to-save-cost-improve-efficiency-overall-profitability-with-trames.md (hunk 1 above)
Comment (ryanseee) [reply cms/event/digitalise-your-supply-chains-to-save-cost-improve-efficiency-overall-profitability-with-trames.md]: done.

**Review by simboonlong — APPROVED** [review]
lgtm, maybe can have some UI text when filter returns nothing (eg. no items found)

**Pull request merged**